### PR TITLE
Pressroom releases

### PR DESCRIPTION
**Pull request by DiogoSoaress**
## What it solves
Adds the Press releases section on the Pressroom page
Refactors the quick categories filter to its own component in the `/common` folder

## Screenshots
Screenshot 2024-03-25 at 11 02 40

## Figma
https://www.figma.com/file/zNUgOvD1WcFankfXfBiAfs/Press-room?type=design&node-id=9929-4249&mode=design&t=x13Zf5TNtfe0ytJN-0

### CODE REVIEW

File: src/components/common/CategoryFilter/index.tsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        return (
+          <Grid item key={category} className={css.filterCard} xs={12} md="auto">
+            <ButtonBase
+              className={clsx(css.filterButton, { [css.selected]: isSelected })}
+              onClick={() => handleToggleCategory(category)}
+            >
+              <Typography>
+                {category}
+                {isSelected && (
+                  <IconButton className={css.closeFilter} onClick={() => handleToggleCategory(category)}>
```
Comment (iamacook) [inline src/components/common/CategoryFilter/index.tsx:42]: Nit: we can remove the duplication by creating a handler for `() => handleToggleCategory(category)`.

File: src/components/common/ShowMoreButton/index.tsx [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import { Box, Button } from '@mui/material'
+import { type NextRouter } from 'next/router'
+import NextLink from 'next/link'
+
+const PAGE_QUERY_PARAM = 'page'
+
+export const getPage = (query: NextRouter['query']): number => {
```
Comment (iamacook) [inline src/components/common/ShowMoreButton/index.tsx:7]: Nit: I'd suggest moving this to a utility file as it is used in multiple components.

Comment (DiogoSoaress) [reply src/components/common/ShowMoreButton/index.tsx]: Thanks for catching this! I wanted to extract it to an util function but forgot.

File: src/components/common/CategoryFilter/index.tsx [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        return (
+          <Grid item key={category} className={css.filterCard} xs={12} md="auto">
+            <ButtonBase
+              className={clsx(css.filterButton, { [css.selected]: isSelected })}
+              onClick={handleToggleCategory(category)}
+            >
+              <Typography>
+                {category}
+                {isSelected && (
+                  <IconButton className={css.closeFilter} onClick={handleToggleCategory(category)}>
```
Comment (iamacook) [inline src/components/common/CategoryFilter/index.tsx:43]: Currying this still creates the function twice. You can create it within the `map` once, e.g.
```suggestion
  const handleToggleCategory = (category: string) => {
    const queryParams = { ...router.query }

    if (queryParams.category === category) {
      delete queryParams.category
    } else {
      queryParams.category = category
    }

    router.push(
      {
        query: queryParams,
      },
      undefined,
      { shallow: true },
    )
  }

  return (
    <Grid container className={css.filterWrapper}>
      {categories.map((category) => {
        const isSelected = category === selectedCategory
        const handleToggle = () => handleToggleCategory(category)

        return (
          <Grid item key={category} className={css.filterCard} xs={12} md="auto">
            <ButtonBase
              className={clsx(css.filterButton, { [css.selected]: isSelected })}
              onClick={handleToggle}
            >
              <Typography>
                {category}
                {isSelected && (
                  <IconButton className={css.closeFilter} onClick={handleToggle}>
```

Comment (DiogoSoaress) [reply src/components/common/CategoryFilter/index.tsx:43]: WDYT of memoizing the function with `useCallback` ?

Comment (DiogoSoaress) [reply src/components/common/CategoryFilter/index.tsx]: Moved the handler inside of the mapping function

**Review by iamacook — APPROVED** [review]

**Pull request merged**